### ISSUE TITLE
Bug in static_methods

### ISSUE THREAD

**Issue by MasonProtter** [oxinabox/Tricks.jl#23]
Run on version 1.6 beta 1:
```julia
julia> using Tricks

julia> let
           f(x) = x + 1
           f(::Int) = 1
           static_methods(f)
       end
# 2 methods:
[1] (::var"#f#7")(::Int64) in Main at REPL[7]:3
[2] (::var"#f#7")(x) in Main at REPL[7]:2

julia> let
           f(x) = x + 1
           @show (length ∘ collect ∘ static_methods)(f)
           f(::Int) = 1
           static_methods(f)
       end
((length ∘ collect) ∘ static_methods)(f) = 1
# 1 method:
[1] (::var"#f#8")(x) in Main at REPL[8]:2
```
The related tests are also failing for me locally.

```julia
julia> versioninfo()
Julia Version 1.6.0-beta1
Commit b84990e1ac* (2021-01-08 12:42 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Ryzen 5 2600 Six-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.0 (ORCJIT, znver1)
Environment:
  JULIA_NUM_THREADS = 6


(@v1.6) pkg> st Tricks
Status `~/.julia/environments/v1.6/Project.toml`
  [410a4b4d] Tricks v0.1.3

(@v1.6) pkg> test Tricks
     Testing Tricks
Status `/tmp/jl_JV12np/Project.toml`
  [410a4b4d] Tricks v0.1.3
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [8dfed614] Test `@stdlib/Test`
Status `/tmp/jl_JV12np/Manifest.toml`
  [410a4b4d] Tricks v0.1.3
  [2a0f44e3] Base64 `@stdlib/Base64`
  [b77e0a4c] InteractiveUtils `@stdlib/InteractiveUtils`
  [56ddb016] Logging `@stdlib/Logging`
  [d6f4376e] Markdown `@stdlib/Markdown`
  [9a3f8284] Random `@stdlib/Random`
  [9e88b42a] Serialization `@stdlib/Serialization`
  [8dfed614] Test `@stdlib/Test`
     Testing Running tests...
Test Summary:    | Pass  Total
static_hasmethod |   35     35
Test Summary:    | Pass  Total
compat_hasmethod |    1      1
static_methods: Test Failed at /home/mason/.julia/packages/Tricks/Omjx9/test/runtests.jl:89
  Expression: ((length ∘ collect) ∘ static_methods)(f) == 2
   Evaluated: 1 == 2
Stacktrace:
 [1] macro expansion
   @ ~/.julia/packages/Tricks/Omjx9/test/runtests.jl:89 [inlined]
 [2] macro expansion
   @ ~/julia/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1147 [inlined]
 [3] top-level scope
   @ ~/.julia/packages/Tricks/Omjx9/test/runtests.jl:86
Test Summary:  | Pass  Fail  Total
static_methods |    7     1      8
ERROR: LoadError: Some tests did not pass: 7 passed, 1 failed, 0 errored, 0 broken.
in expression starting at /home/mason/.julia/packages/Tricks/Omjx9/test/runtests.jl:84
ERROR: Package Tricks errored during testing
```

**Comment by oxinabox** [comment]
Does this work on 1.5?

**Comment by oxinabox** [comment]
It looks like the back edge is not triggering when  `f(::Int) = 1` is run.

**Comment by MasonProtter** [comment]
Yep, works on 1.5 for me:
```julia
julia> using Tricks

julia> let
           f(x) = x + 1
           f(::Int) = 1
           static_methods(f)
       end
# 2 methods:
[1] (::var"#f#3")(::Int64) in Main at REPL[6]:3
[2] (::var"#f#3")(x) in Main at REPL[6]:2

julia> let
           f(x) = x + 1
           @show (length ∘ collect ∘ static_methods)(f)
           f(::Int) = 1
           static_methods(f)
       end
((length ∘ collect) ∘ static_methods)(f) = 1
# 2 methods:
[1] (::var"#f#4")(::Int64) in Main at REPL[7]:4
[2] (::var"#f#4")(x) in Main at REPL[7]:2

julia> versioninfo()
Julia Version 1.5.3
Commit 788b2c77c1* (2020-11-09 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Ryzen 5 2600 Six-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, znver1)
Environment:
  JULIA_NUM_THREADS = 6

(@v1.5) pkg> st
Status `~/.julia/environments/v1.5/Project.toml`
  [410a4b4d] Tricks v0.1.3
```

**Comment by oxinabox** [comment]
Hmm might have to learn to use `git bisect`

**Comment by MasonProtter** [comment]
We should probably track this down before 1.6 is released in case it's a real bug in julia. I don't know anything about git bisect, do you have any pointers?

**Comment by oxinabox** [comment]
I also don't know anything about it.
Except that it can be used for finding which commit caused a test to fail

**Comment by MasonProtter** [comment]
Okay, reading the docs it doesn't see, too hard. I'll it now.

**Comment by MasonProtter** [comment]
Nvm, this is fixed now on v1.6.0-rc1.
